### ISSUE TITLE
Feature extraction GUI update

### ISSUE THREAD

**Issue by alex4200** [cnr-ibf-pa/hbp-bsp-issues#334]
## Use Case Update
 
Aspect | Detail
------------ | -------------
Summary |  the current feature extraction web app will be modified in order to fetch the electrophysiological data (from which the feature are extracted) from the HBP dataset (container/KG). At the moment the data are stored (and fetched from) the web-app backend storage.
Expert | @lbologna 
Deadline |  Expected date of the update first release: TBD. It depends on the curation process and implementation of the new API. A rough estimate at the moment is M7 (next HBP Summit)
Dependencies | 

 
**Remove anything that does not apply**
## List of additional/changed features 
*What are the changed or new features? What activities can users perform while using it? What makes it different from similar use cases? List the main functions that you will build into your product here. Also specify the priority ‘mandatory’, ‘important, ‘nice to have’.*
 
## Acceptance Tests
*Define here the acceptance tests to evaluate the use case’s compliance with the requirements as defined above. Also possible end users for testing can be included here.*
 
### System
*Do the changes require any additional system requirements (any special software package)?*
 
### Performance
*What are additional performance requirements for the features? What time should it take to run them?*

**Comment by antonelepfl** [comment]
Working on adapting the WebUI to be able to read the format of KG

**Comment by lbologna** [comment]
The BluePyEfe software, used for the feature extraction, reads the following format: .abf, .smr, .igor, .txt (with metadata).
At the moment, only a limited number of .abf traces is available in the KG (no .smr, no .igor, .txt to be checked) that have to be tested and are different from the ones available at the moment on the backend server. 
For the moment we will leave the current status (data on the backend).

By M12 we will implement a first version (dev) of the needed interface to fetch a subset of .abf files available on the KG, provided that these files have the needed requirements (step currents stimuli, .abf v2 files).

**Comment by alex4200** [comment]
This ticket will be probably postponed to M18

**Comment by lbologna** [comment]
Postponed to M18

**Comment by alex4200** [comment]
@lbologna Please do not forget to update this task with some more details

**Comment by lbologna** [comment]
> @lbologna Please do not forget to update this task with some more details

That's done (in the main issue description).

**Comment by apdavison** [comment]
Are you aware that we are developing in SP5 a Python library for querying and fetching data from the KG? I suggest we have a video meeting to discuss the requirements.

**Comment by lbologna** [comment]
> Are you aware that we are developing in SP5 a Python library for querying and fetching data from the KG? I suggest we have a video meeting to discuss the requirements.

@apdavison no, I was not aware you were on it. Yes, for sure a call would be very useful. What about Thursday (23 May) or Friday (24 May) any time before 13? Thanks

**Comment by apdavison** [comment]
How about Friday 24th at 12 noon?

**Comment by lbologna** [comment]
Fine with us! Thanks.

**Comment by apdavison** [comment]
Update: the NAR ("Neural Activity Resource") library has now been released under the name "fairgraph" - https://github.com/HumanBrainProject/fairgraph (also available with `pip install fairgraph`). This can be used to retrieve models and data from the KG - see tutorial notebook at https://collab.humanbrainproject.eu/#/collab/1635/nav/380436

**Comment by alex4200** [comment]
@apdavison I have run the example (command line and notebook), but in the section **An activity dataset with minimal metadata** it does not find any dataset. 

Can you look into that and either fix it, or update README.md and the notebook?

**Comment by clupascu** [comment]
@apdavison For the 26 cells retrieved based on the brain region ("hippocampus CA1") 
cells_in_ca1 = PatchedCell.list(client, brain_region=BrainRegion("hippocampus CA1"))
is there any possibility to use one other filter in order to retrieve one specific cell, instead of doing 
example_cell = cells_in_ca1[3] ?

I see no other metadata apart for the CellType and BrainRegion

PatchedCell('hbp00011_Sub2_Samp4__ExpE8', CellType('hippocampus CA1 pyramidal cell', 'http://uri.neuinfo.org/nif/nifstd/sao830368389'), BrainRegion('hippocampus CA1', 'http://purl.obolibrary.org/obo/UBERON_0003881'), KGQuery(<class 'fairgraph.electrophysiology.PatchedCellCollection'>, {'path': 'prov:hadMember', 'op': 'in', 'value': ['https://nexus.humanbrainproject.org/v0/data/neuralactivity/experiment/patchedcell/v0.1.0/75c761ba-ce25-4659-aa2d-c4b6e2a417fc']}), https://nexus.humanbrainproject.org/v0/data/neuralactivity/experiment/patchedcell/v0.1.0/75c761ba-ce25-4659-aa2d-c4b6e2a417fc)

Thanks.

**Comment by clupascu** [comment]
> @apdavison I have run the example (command line and notebook), but in the section **An activity dataset with minimal metadata** it does not find any dataset.
> 
> Can you look into that and either fix it, or update README.md and the notebook?

@alex4200 in the section **An activity dataset with minimal metadata**  I get 29 datasets.

**Comment by apdavison** [comment]
> is there any possibility to use one other filter in order to retrieve one specific cell

If you know the cell id (either UUID or full URI) or the cell name, you can retrieve it with `PatchedCell.from_uuid(uuid)`, `PatchedCell.from_uri(uri)` or `PatchedCell.by_name(name)`.

Using `PatchedCell.list()`, only one filter at a time is currently allowed. I've created a ticket to fix this: https://github.com/HumanBrainProject/fairgraph/issues/9

Can you tell me exactly what properties you need to filter on?

> I see no other metadata apart for the CellType and BrainRegion

There are many other metadata attributes, see https://github.com/HumanBrainProject/fairgraph/blob/master/fairgraph/electrophysiology.py#L136

If information you need is missing, please let me know so we can add it!

**Comment by apdavison** [comment]
@alex4200 do you still get an empty list? If so, it may be a permissions problem. I presume you have an SGA2 accreditation? (https://collab.humanbrainproject.eu/#/me/accreditations)

**Comment by clupascu** [comment]
> 
> 
> > is there any possibility to use one other filter in order to retrieve one specific cell
> 
> If you know the cell id (either UUID or full URI) or the cell name, you can retrieve it with `PatchedCell.from_uuid(uuid)`, `PatchedCell.from_uri(uri)` or `PatchedCell.by_name(name)`.
> 
> Using `PatchedCell.list()`, only one filter at a time is currently allowed. I've created a ticket to fix this: [HumanBrainProject/fairgraph#9](https://github.com/HumanBrainProject/fairgraph/issues/9)

More than one filter would be needed. Thanks.

> 
> Can you tell me exactly what properties you need to filter on?
> 
> > I see no other metadata apart for the CellType and BrainRegion
> 
> There are many other metadata attributes, see https://github.com/HumanBrainProject/fairgraph/blob/master/fairgraph/electrophysiology.py#L136
> 
> If information you need is missing, please let me know so we can add it!

Can I retrieve cells based on the contributor name or the lab for example?

**Comment by lbologna** [comment]
Hi,
@apdavison I am experiencing the same problem @alex4200 has: empty lists for all searches (based on brain region, based on species, based on cell type). I have the SGA2 accreditation.

**Comment by alex4200** [comment]
@apdavison I have the SGA1 and SGA2 accreditation. I just tried the query again in a notebook and from the command line, but still get an empty list only.

**Comment by apdavison** [comment]
@lbologna and @alex4200 I've changed the permissions, please could you try again?

**Comment by lbologna** [comment]
@apdavison thank you. It works for me but only up till the "Search based on cell type" cell.
The "**An activity dataset with minimal metadata**" and "**An activity dataset with extended metadata**" section seem to fetch nothing and give the errors below:

- Error for "An activity dataset with minimal metadata"


IndexError                                Traceback (most recent call last)

----> 1 dataset = activity_datasets[-1]
      2 dataset.owners[0].resolve(client)

IndexError: list index out of range

------------------------------------------------

- Error for "An activity dataset with extended metadata"

AttributeError                            Traceback (most recent call last)
      5     "path": "nsg:partOf",
      6     "op": "eq",
----> 7     "value": dataset.id
      8 }
      9 context = {

AttributeError: 'NoneType' object has no attribute 'id'

**Comment by alex4200** [comment]
@apdavison Have you looked into this issue yet?

**Comment by apdavison** [comment]
I'm looking into it; waiting for some feedback from the KG team.
